pkg/gofr/datasource/mongo/mongo.go

```diff
@@ -108,18 +113,60 @@ func (c *Client) Connect() {
 	}
 
 	if err = m.Ping(ctx, nil); err != nil {
-		c.logger.Errorf("could not connect to mongoDB at %v due to err: %v", c.config.URI, err)
+		c.logger.Errorf("could not connect to mongoDB at %v due to err: %v", host, err)
 		return
 	}
 
-	c.logger.Logf("connected to mongoDB successfully at %v to database %v", c.config.URI, c.config.Database)
+	c.logger.Logf("connected to mongoDB successfully at %v to database %v", host, c.config.Database)
 
 	mongoBuckets := []float64{.05, .075, .1, .125, .15, .2, .3, .5, .75, 1, 2, 3, 4, 5, 7.5, 10}
 	c.metrics.NewHistogram("app_mongo_stats", "Response time of MONGO queries in milliseconds.", mongoBuckets...)
 
 	c.Database = m.Database(c.config.Database)
 
-	c.logger.Logf("connected to MongoDB at %v to database %v", uri, c.Database)
+	c.logger.Logf("connected to MongoDB at %v to database %v", host, c.Database)
+}
+
+func generateMongoURI(config *Config) (uri, host string, err error) {
+	if config.URI != "" {
+		host, err = getDBHost(config.URI)
+		if err != nil || host == "" {
+			return "", "", err
+		}
+
+		return config.URI, host, nil
+	}
+
+	switch {
+	case config.Host == "":
+		return "", "", fmt.Errorf("%w: host is empty", errMissingField)
+	case config.Port == 0:
+		return "", "", fmt.Errorf("%w: port is empty", errMissingField)
+	case config.Database == "":
+		return "", "", fmt.Errorf("%w: database is empty", errMissingField)
+	}
+
+	uri = fmt.Sprintf("mongodb://%s:%s@%s:%d/%s?authSource=admin",
+		config.User, config.Password, config.Host, config.Port, config.Database)
```

Comment:
Be careful here

This approach could lead to bugs

user and password (but more likely the password can contain character that needs to be escaped, such as @#: it would lead to invalid URL, or password that won't be readable by mongodb client

They should be escaped I'm unsure what is the RFC for this.

Also the database may contains slash or ? (I'm usure it's possible with mongoDB), so here url.PathEscape should be used for  comfig.Database

I would like to suggest creating a url.URL with all required fields then use the .String

Please note, this could be done in a separate PR as the parttern of forging URL by concatenation might be used somewhere else